### PR TITLE
feat(azure-iot-mqtt-device): modelID option

### PR DESCRIPTION
**Pull request by YoDaMa**
This adds the setOption for ModelID, which will enable users to use the existing device client library as a PnP library. To accomodate the lack of support in the existing service API, a switching API call has been added that will use the preview API version if the modelID is set.

### CODE REVIEW

File: device/core/src/interfaces.ts [hunk 1]
```diff
@@ -129,6 +129,12 @@ export interface DeviceClientOptions extends X509 {
    */
   productInfo?: string;
 
+  /**
+   * !!Digital Twin Use Only!!
+   * String used in MQTT username settting the Digital Twin modelID.
```
Comment (zolvarga) [inline device/core/src/interfaces.ts:134]: Typo

Comment (zolvarga) [inline device/core/src/interfaces.ts:134]: modelId

File: device/transport/mqtt/devdoc/mqtt_requirements.md [hunk 2]
```diff
@@ -225,6 +227,9 @@ The `reject` method is there for compatibility purposes with other transports bu
 **SRS_NODE_DEVICE_MQTT_06_002: [** The authentication providers `setTokenRenewalValues` method shall be invoked with the values provided in the tokenRenewal option.
  **]**
 
+**SRS_NODE_DEVICE_MQTT_41_014: [** For a Plug and Play Device the modelID should be included as `&digital-twin-model-id=<DEVICE’s MODEL ID>` after the api-version **]**
```
Comment (zolvarga) [inline device/transport/mqtt/devdoc/mqtt_requirements.md:230]: Same...

File: device/transport/mqtt/src/mqtt.ts [hunk 3]
```diff
@@ -727,14 +732,18 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
       clientId = credentials.deviceId;
     }
 
+    /*Codes_SRS_NODE_DEVICE_MQTT_41_015: [If a modelID is provided, the device should use the PnP API String] */
```
Comment (zolvarga) [inline device/transport/mqtt/src/mqtt.ts:735]: >modelID [](start = 47, length = 7)

Same...

Comment (zolvarga) [inline device/transport/mqtt/src/mqtt.ts:735]: >modelID [](start = 47, length = 7)

Same...

File: device/transport/mqtt/test/_mqtt_test.js [hunk 4]
```diff
@@ -634,6 +634,31 @@ describe('Mqtt', function () {
       });
     });
 
+    /* Tests_SRS_NODE_DEVICE_MQTT_41_014: [For a Plug and Play Device the modelID should be included as `&digital-twin-model-id=<DEVICE’s MODEL ID>` after the api-version]*/
```
Comment (zolvarga) [inline device/transport/mqtt/test/_mqtt_test.js]: >modelID [](start = 74, length = 7)

Same...

File: device/transport/mqtt/test/_mqtt_test.js [hunk 5]
```diff
@@ -634,6 +634,31 @@ describe('Mqtt', function () {
...
+    /* Tests_SRS_NODE_DEVICE_MQTT_41_015: [If a modelID is provided, the device should use the PnP API String] */
```
Comment (zolvarga) [inline device/transport/mqtt/test/_mqtt_test.js]: >modelID [](start = 48, length = 7)

Same...

File: device/core/src/interfaces.ts (hunk 1 above)
Comment (anthonyvercolano) [reply device/core/src/interfaces.ts]: Well if we're gonna get picky, there are three 't''s in setting

---
In reply to: [431961220](https://github.com/Azure/azure-iot-sdk-node/pull/809#discussion_r431961220) [](ancestors = 431961220)

**Pull request merged**